rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,108 @@
+---
+feature: NixOS Code of Conduct
+start-date: 2021-11-03
+author: @jonringer
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues: #98 #102
+---
+
+# Summary
+[summary]: #summary
+
+The NixOS Community needs some foundation of shared values which can be used
+to determine appropriate when contributing and interacting with the community.
+We currently have a [community ethos](https://nixos.org/community/index.html), but
+this only states that we are an inclusive community, but doesn't define
+inappropriate behavior when interacting with the community. This RFC aims
+to establish the shared values, so they can be used to identify behavior
+which is disruptive to the community.
+
+This RFC does not intend to define moderation practices. This is intended to only lay
+a framework of shared values for the community.
+
+# Motivation
+[motivation]: #motivation
+
+The NixOS community struggles with a clearly defined way in which someone's behavior may
+be disruptive. A well-defined Code of Conduct gives an explicit
+set of expectations for all contributors, which will also make it easier to identify
+when behavior becomes disruptive.
+
+# Detailed design
+[design]: #detailed-design
+
+These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
+which provides a great compromise between explicit behaviors and subjective goals.
+Only the moderation body name has been changed.
+
+- We are committed to providing a friendly, safe and welcoming environment for
+all, regardless of level of experience, gender identity and expression,
```

Comment:
I have a feeling that one of the attributes is not like the others here, and pretending it is will only lead to more events that feel hypocritical to some. Basically, we want the environment to be safe and friendly regardless of level of experience, but at some level it is very much not welcoming in practice, because the environment where all conversations go above your head is not too welcoming. And whatever idealists say, Nix is complicated enough that Nix spaces will be a bad place to learn basic Linux knowledge for most people.

While I agree with @asymmetric about the two additions, I guess having access to something powerful enough to evaluate anything useful is an economic prerequisite to the place succeeding at being welcoming.

Reply:
> we want the environment to be safe and friendly regardless of level of experience, but at some level it is very much not welcoming in practice, because the environment where all conversations go above your head is not too welcoming.

This sounds more related to having a "noob friendly" area, rather an issue with personal conduct. This is to just to make it clear that people with little experience are also welcome.

> And whatever idealists say, Nix is complicated enough that Nix spaces will be a bad place to learn basic Linux knowledge for most people.

I agree, but I'm not sure what the correlation is with this CoC. Learning materials has always been a pain point in nix's past.

Reply:
I have seen bans that to me looked like bans for being too clueless… Our noobs do have quite a bit background, and learn with reasonable speed or go away frustrated. And the learning materials we are hoping for — they would still start rather high in terms of expected backgrounds (which I find reasonable).

Maybe even I would qualify «all» with «all willing to learn»… It's true that newbies who are below our level but ask how to learn have a reasonable chance to get a recommendation for path to filling in missing background (probably somewhere outside) — and when I say chance, I mostly mean that recommending learning paths is hard, sometimes it doesn't happen or a recommendation is suboptimal.

Reply:
If someone is being target just because they are new, then obviously that's an issue. But I think the current situation would be improved by just having better on-boarding documentation. Nix pills is a great second resource, but it does a very deep dive into how nix works, but doesn't really help with most nix use cases.

My personal efforts with @nrdxp are to create the "nix book" https://book.divnix.com/ . However, it keeps getting delayed because of other nix related events, like this.

Reply:
Well, they are banned because of their level of cluelessness is declared disruptive (with significant reasons for that declaration, sure, whether sufficient or not). 

Maybe I want to qualify this in some way because the other listed attributes should not even affect the substantial conversation except the relatively special cases («as a colourblind contributor X finds the choices in the markup frustrating» or maybe the converse «you have 100% eyesight sharpness, so I am not surprised you find this font size optimal, but not everyone is like that»), and offtopic conversations also only in similar circumstances. Being inexperienced puts some expectations on the person about learning and maybe even not jumping ahead too much.

Reply:
> Well, they are banned because of their level of cluelessness is declared disruptive (with significant reasons for that declaration, sure, whether sufficient or not).

I don't have context on this, cc @grahamc for thoughts.

Reply:
A community made more inclusive by…
At that, why is that this one immediately singles people out for their immutable characteristics, rather than include them?